mlir/include/mlir/Dialect/ArmSME/IR/ArmSMEOps.td

```diff
@@ -768,4 +783,33 @@ let arguments = (ins
   }];
 }
 
+def StreamingVLOp : ArmSME_Op<"streaming_vl", [Pure]>
+{
+  let summary = "Query the streaming vector length";
+
+  let description = [{
+    This operation returns the streaming vector length for a given type size.
+    Unlike `vector.vscale` the value returned is invariant to the streaming
+    mode.
+
+    Example:
+    ```mlir
+    // Streaming vector length in:
+    // - bytes (8-bit)
+    %svl_b = arm_sme.streaming_vl <byte>
+    // - half words (16-bit)
+    %svl_h = arm_sme.streaming_vl <half>
+    // - words (32-bit)
+    %svl_w = arm_sme.streaming_vl <word>
+    // - double words (64-bit)
+    %svl_d = arm_sme.streaming_vl <double>
+    ```
```

Comment:
For consistency with the spec:

```suggestion
  let description = [{
    This operation returns the streaming vector length (SVL) for a given type size.
    Unlike `vector.vscale` the value returned is invariant to the streaming
    mode.

    Example:
    ```mlir
    // Streaming vector length in:
    // - bytes (8-bit, SVLb)
    %svl_b = arm_sme.streaming_vl <byte>
    // - half words (16-bit, SVLh)
    %svl_h = arm_sme.streaming_vl <half>
    // - words (32-bit, SVLw)
    %svl_w = arm_sme.streaming_vl <word>
    // - double words (64-bit, SVLd)
    %svl_d = arm_sme.streaming_vl <double>
    ```
```